processor/src/fast/step.rs

```diff
@@ -0,0 +1,77 @@
+//! This module defines items relevant to controlling execution stopping conditions, through the
+//! `FastProcessor::step()` method.
+
+use alloc::sync::Arc;
+
+use miden_core::{Kernel, mast::MastForest};
+
+use crate::{
+    ExecutionError,
+    continuation_stack::{Continuation, ContinuationStack},
+    fast::FastProcessor,
+};
+
+// RESUME CONTEXT
+// ===============================================================================================
+
+/// The context required to resume execution of a program from the last point at which it was
+/// stopped.
+#[derive(Debug)]
+pub struct ResumeContext {
```

Comment:
I see you mentioned the debugger API is incomplete. The fields here are public, but figuring out "what operation executes next" from the continuation stack seems tricky. `ResumeBasicBlock` requires looking up the operation in the forest, while other continuations like `AfterExitDecorators` don't map to operations at all.

Would a helper method make sense here? Something like:

```rust
impl ResumeContext {
    pub fn next_operation(&self) -> Option<OperationInfo> {
        // ...
    }
}
```

Just thinking this might make things easier for debugger authors.